stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -59,6 +61,7 @@ def _run_split_procurement(self, procurements):
         precision = self.env["decimal.precision"].precision_get(
             "Product Unit of Measure"
         )
+        done_procurements_by_product = {}
```

Comment:
```suggestion
        qty_unavailable_by_product = {}
```

Reply:
I would also suggest to use @jbaudoux naming, because this more precise. 
But i would switch to a `defaultdict` this would ease the later calls. 
```suggestion
        done_procurements_by_product = defaultdict(float)
```